docs/content/Switch.mdx

```diff
@@ -0,0 +1,225 @@
+    </>
+  )
+}
+
+render(Controlled)
+```
+
+### Small
+
+```jsx live
+<Box display="flex" maxWidth="300px">
+  <Box flexGrow={1} fontSize={1} fontWeight="bold" id="switchLabel">
+    Notifications
+  </Box>
+  <Switch aria-labelledby="switchLabel" size="small" />
+</Box>
+```
+
+### Delayed toggle with loading state
+
+```javascript noinline live
+const LoadingToggle = () => {
+  const [isLoading, setIsLoading] = React.useState(false)
+  const [isOn, setIsOn] = React.useState(false)
+
+  async function switchSlowly(currentOn) {
+    await new Promise(resolve => setTimeout(resolve, 1500))
+    return await !currentOn
+  }
+
+  async function onClick() {
+    setIsLoading(!isLoading)
+    const newSwitchState = await switchSlowly(isOn)
+    setIsOn(newSwitchState)
+  }
+
+  const handleSwitchChange = React.useCallback(
+    on => {
+      setIsLoading(false)
+    },
+    [setIsLoading]
+  )
+
+  return (
+    <>
+      <Box display="flex" maxWidth="300px">
+        <Box flexGrow={1} fontSize={2} fontWeight="bold" id="switchLabel">
+          Notifications
+        </Box>
+        <Switch
+          aria-labelledby="switchLabel"
+          isLoading={isLoading}
+          on={isOn}
+          onClick={onClick}
+          onChange={handleSwitchChange}
+        />
+      </Box>
+      <p>The switch is {isOn ? 'on' : 'off'}</p>
+    </>
+  )
+}
+
+render(LoadingToggle)
+```
+
+### Disabled
+
+```jsx live
+<Box display="flex" maxWidth="300px">
+  <Box flexGrow={1} fontSize={2} fontWeight="bold" id="switchLabel">
+    Notifications
+  </Box>
+  <Switch aria-labelledby="switchLabel" disabled />
+</Box>
+```
+
+### With associated caption text
+
+```jsx live
+<Box display="flex">
+  <Box flexGrow={1}>
+    <Text fontSize={2} fontWeight="bold" id="switchLabel" display="block">
+      Notifications
+    </Text>
+    <Text color="fg.subtle" fontSize={1} id="switchCaption" display="block">
+      Notifications will be delivered via email and the GitHub notification center
+    </Text>
+  </Box>
+  <Switch aria-labelledby="switchLabel" aria-describedby="switchCaption" />
+</Box>
+```
+
+### Left-aligned with label
+
+```jsx live
+<>
+  <Text fontSize={2} fontWeight="bold" id="switchLabel" display="block" mb={1}>
+    Notifications
+  </Text>
+  <Switch statusLabelPosition="end" aria-labelledby="switchLabel" />
+</>
+```
+
+## Props
+
+/\*\* Whether the "on" and "off" labels should appear before or after the switch.
+
+- **This should only be changed when the switch's alignment needs to be adjusted.** For example: It needs to be left-aligned because the label appears above it and the caption appears below it.
+  \*/
+  statusLabelPosition?: 'start' | 'end'
+
+<PropsTable>
+  <PropsTableRow name="aria-describedby" type="string" description="The id of the DOM node that describes the switch" />
+  <PropsTableRow
+    name="aria-labelledby"
+    type="string"
+    required
+    description="The id of the DOM node that labels the switch"
+  />
```

Comment:
What's the difference between "labeling" and "describing"?

Reply:
The label identifies what the switch is, and the description gives more info about what the switch affects.

For example, if the [wiki setting](https://user-images.githubusercontent.com/2313998/157947699-e2029327-6cd8-4841-b1ab-532719c65daf.png) was converted to use a switch instead of a checkbox, "Wikis" would be the label, and "Wikis host documentation for your repository."

Reply:
Ah makes sense. Thank you!